lib/sensors/gyroscope.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "i2c_sensors.hpp"
+
+#include <cstdint>
+#include <optional>
+
+#include <core/logger.hpp>
+#include <io/i2c.hpp>
+
+namespace hyped::sensors {
+
+// Registers taken from the data sheet
+static constexpr std::uint8_t kDefaultGyroscopeAddress = 0x69;
+static constexpr std::uint8_t kDataXHigh               = 0x29;
+static constexpr std::uint8_t kDataXLow                = 0x28;
+static constexpr std::uint8_t kDataYHigh               = 0x2B;
+static constexpr std::uint8_t kDataYLow                = 0x2A;
+static constexpr std::uint8_t kDataZHigh               = 0x2D;
+static constexpr std::uint8_t kDataZLow                = 0x2C;
+static constexpr std::uint8_t kCtrl1                   = 0x20;
+static constexpr std::uint8_t kCtrl2                   = 0x21;
+static constexpr std::uint8_t kCtrl3                   = 0x22;
+static constexpr std::uint8_t kCtrl5                   = 0x24;
+static constexpr std::uint8_t kConfigurationSetting1   = 0xff;
```

Comment:
I know we usually use full names when in doubt but `kConfigurationSetting1` seems quite verbose. Could we shorten this to `kConfigSetting1` since don't think this harms readability as it is a common abbreviation?